snapcraft/plugins/kernel.py

```diff
@@ -142,16 +141,14 @@ def __init__(self, name, options):
         self.make_install_targets.extend(self._get_fw_install_targets())
 
         self.os_snap = os.path.join(self.sourcedir, 'os.snap')
-        self._target_arch = get_machine_info(target_machine)
 
-    def set_target_machine(self, machine):
-        self._target_arch = get_machine_info(machine)
+    def set_target_machine(self):
```

Comment:
> The problem is plugins that don't support cross compiling.

You mean as of right now if one tries cross-compiling with plugins that don't support it you get a good error? That's an excellent point.